src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java

```diff
@@ -169,6 +175,21 @@ public JSONArray getBuildCauses(String className) throws IOException, ClassNotFo
         return result;
     }
 
+    private boolean isClassOrSubclass(Class<?> clazz, String classNameToCheck) {
+        String causeClassName = clazz.getName();
+        if (causeClassName.equals("java.lang.Object")) {
+            return false;
+        } else if (causeClassName.equals(classNameToCheck)) {
+            return true;
+        } else {
+            if (causeClassName.contains("$")) {
```

Comment:
if the cause is foo.bar.Wibble$Cause would you not expect the user to enter that?  Are there any instances of this, worth writing a test for?

Reply:
I'll look into writing a test but i think its already covered. Take the case of `hudson.build.Cause$UserCause` - if the user wants to check to see if a cause is of that type, they have to enter that string.  I found that `clazz.getClass().getSuperClass().getName()` was returning `java.lang.Object` when it was passed to the next level of recursion (which makes sense i believe) which is why i added the check to pass `clazz.getEnclosingClass()` when needed.

Reply:
I'm also confused by this check. It seems like you'd never want to check the enclosing class, since that is just about source-level organization, not about the actual sub/superclass relationships which should be handled by the recursive calls to `getSuperclass()`. 

> It is a bit odd to me though, because class.getClass().getSuperClass().getName() returns the enclosing class when evaluated in the ide


In the case of `hudson.build.Cause$UserCause`, `UserCause` directly extends `hudson.build.Cause`, so `hudson.build.Cause` is both the superclass and the enclosing class. Is that what was happening?